File: prendreModule.js
const prompt = require("prompt-sync")();
const walletModule = require("./walletModule");
const objetElement = {
  C1: "CIN",
  C2: "CB",
  C3: "CV",
  C4: "PHOTO",
  C5: "Argent",
};

function prendreElements(wallet) {
  console.log("Options pour prendre des éléments du wallet :");
  console.log("[1] Prendre une carte d'identité (CIN)");
  console.log("[2] Prendre une carte bancaire (CB)");
  console.log("[3] Prendre un carte de visite (CV)");
  console.log("[4] Prendre une photo (PHOTO)");
  console.log("[5] Prendre de l'argent");
  console.log("[6] Revenir au menu principal");

  let choix = prompt("Veuillez sélectionner une option (1-6) : ");

  switch (choix) {
    case "1":
    case "2":
    case "3":
    case "4":
      prendreElement(choix, wallet); // Utilisez getWallet() sans les parenthèses
      break;
    case "5":
      prendreArgent();
      break;
    case "6":
      walletModule.gestionOptions; // Utilise gestionOptions au lieu de prendreElements
      return;
    default:
      console.log("Option invalide.");
  }

  prendreElements();
}

function prendreElement(type, wallet) {
  const cle = objetElement[`C${type}`];
  if (wallet && wallet[cle] > 0) {
    wallet[cle] = 0;
    console.log(`Vous avez pris une ${cle}.`);
  } else {
    console.log(`Le wallet ne contient pas de ${cle}.`);
  }
}

function prendreArgent() {
  let montant = prompt(
    "Veuillez saisir la somme d'argent que vous souhaitez prendre : "
  );

  try {
    if (parseFloat(montant) > argent) {
      throw new Error("Fonds insuffisants dans le wallet.");
    }
    argent -= parseFloat(montant);
    console.log(`Vous avez pris ${montant} euros.`);
  } catch (error) {
    console.log(`Erreur : ${error.message}`);
  }
  prendreElements();
}

module.exports = {
  prendreElements,
  prendreElement,
  prendreArgent,
};

File: prendreModule.py
import walletModule

objet_element = {
    "C1": "CIN",
    "C2": "CB",
    "C3": "CV",
    "C4": "PHOTO",
    "C5": "Argent",
}

def prendreElements():
    wallet = walletModule.getWallet()
    print("Options pour prendre des éléments du wallet :")
    print("[1] Prendre une carte d'identité (CIN)")
    print("[2] Prendre une carte bancaire (CB)")
    print("[3] Prendre un carte de visite (CV)")
    print("[4] Prendre une photo (PHOTO)")
    print("[5] Prendre de l'argent")
    print("[6] Revenir au menu principal")

    choix = input("Veuillez sélectionner une option (1-6) : ")

    if choix in ["1", "2", "3", "4"]:
        prendreElement(choix)
    elif choix == "5":
        prendreArgent()
    elif choix == "6":
        walletModule.gestionOptions()
    else:
        print("Option invalide.")
        prendreElements()

def prendreElement(type):
    wallet = walletModule.getWallet()
    cle = f"C{type}"
    if wallet[cle] > 0:
        wallet[cle] = 0
        print(f"Vous avez pris une {cle}.")
    else:
        print(f"Le wallet ne contient pas de {cle}.")

def prendreArgent():
    wallet = walletModule.getWallet()
    montant = float(input("Veuillez saisir la somme d'argent que vous souhaitez prendre : "))
    
    try:
        if montant > wallet["Argent"]:
            raise ValueError("Fonds insuffisants dans le wallet.")
        wallet["Argent"] -= montant
        print(f"Vous avez pris {montant} euros.")
    except ValueError as error:
        print(f"Erreur : {error}")
    prendreElements()
